### PR TITLE
Coursework

### PR DESCRIPTION
**Pull request by askaretc**
EASY LEVEL
1. Employee class has been created.
2. Employee[] array has been created inside the Main class.
3. Static methods printAllEmployeesInformation(), printTotalSalary(), printEmployeeWithMaximalSalary(),  printEmployeeWithMinimalSalary(), printAverageSalary(), printAllEmployeesFullName() have been created.

### CODE REVIEW

**Comment by askaretc** [comment]
MEDIUM LEVEL

Additional static methods indexAllEmployeesSalary(), printEmployeeWithMaximalSalaryByDepartment(), printEmployeeWithMinimalSalaryByDepartment(), printTotalSalaryByDepartment(), printAverageSalaryByDepartment(), indexAllEmployeesSalaryByDepartment(), printAllEmployeesInformationByDepartment(), printEmployeesWithSalaryLessThanNumber(), printEmployeesWithSalaryGreaterThanNumber() have been created.

**Comment by askaretc** [comment]
HARD LEVEL

The project structure has been redesigned to OOP and Anemic model.
1. EmployeeBook class has been created.
2. Employee[] array and static methods have been moved to the Employee Book class.
3. Additional methods addEmployee(), removeEmployee(), editEmployeesSalary(), editEmployeesDepartment() printAllEmployeesFullNameAccordingToDepartment() have been created.

File: src/pro/sky/java/course1/coursework/EmployeeBook.java [hunk 1]
```diff
@@ -0,0 +1,400 @@
...
+                    if (employee.getSalary() == calculateMinimalSalary()) {
+                        System.out.println("Employee with minimal salary: " + employee);
+                    }
+                }
+            }
+        }
+    }
+
+    private int calculateMinimalSalary() {
+        int minimalSalary = 0;
```
Comment (DmitryN10) [inline src/pro/sky/java/course1/coursework/EmployeeBook.java]: int minimalSalary = Integer.MAX_VALUE;

File: src/pro/sky/java/course1/coursework/EmployeeBook.java [hunk 2]
```diff
@@ -0,0 +1,400 @@
...
+        if (checkAllElementsIsNotNull()) {
+            for (Employee employee : employees) {
```
Comment (DmitryN10) [inline src/pro/sky/java/course1/coursework/EmployeeBook.java:186]: тогда тот цикл не нужен будет

File: src/pro/sky/java/course1/coursework/EmployeeBook.java [hunk 3]
```diff
@@ -0,0 +1,400 @@
...
+                    }
+                }
+            }
+        }
+    }
+
+    private int calculateMinimalSalaryByDepartment(int department) {
+        int minimalSalary = 0;
+        if (checkAllElementsIsNotNull()) {
+            for (Employee employee : employees) {
```
Comment (DmitryN10) [inline src/pro/sky/java/course1/coursework/EmployeeBook.java:285]: здесь аналогично можно избавиться от этого цикла

**Pull request merged**